src/saml.ts

```diff
@@ -1077,37 +1077,45 @@ class SAML {
           }
         }
 
-        subjectConfirmation = subject[0].SubjectConfirmation
-          ? subject[0].SubjectConfirmation[0]
-          : null;
-        confirmData =
-          subjectConfirmation && subjectConfirmation.SubjectConfirmationData
-            ? subjectConfirmation.SubjectConfirmationData[0]
-            : null;
-        if (subject[0].SubjectConfirmation && subject[0].SubjectConfirmation.length > 1) {
-          msg = "Unable to process multiple SubjectConfirmations in SAML assertion";
-          throw new Error(msg);
-        }
-
-        if (subjectConfirmation) {
-          if (confirmData && confirmData.$) {
-            const subjectNotBefore = confirmData.$.NotBefore;
-            const subjectNotOnOrAfter = confirmData.$.NotOnOrAfter;
-            const maxTimeLimitMs = this.processMaxAgeAssertionTime(
-              this.options.maxAssertionAgeMs,
-              subjectNotOnOrAfter,
-              assertion.$.IssueInstant
-            );
-
-            const subjErr = this.checkTimestampsValidityError(
-              nowMs,
-              subjectNotBefore,
-              subjectNotOnOrAfter,
-              maxTimeLimitMs
-            );
-            if (subjErr) {
-              throw subjErr;
+        if (subject[0].SubjectConfirmation && subject[0].SubjectConfirmation?.length > 0) {
```

Comment:
Can be simplified to `subject[0].SubjectConfirmation?.length > 0` in current versions of TypeScript. Otherwise, the old syntax can be used: `subject[0].SubjectConfirmation && subject[0].SubjectConfirmation.length > 0`. Either two checks or one `?`; not both.

Reply:
I know; I managed to escape this fate a lot of times, now it's clear to me, I have to LEARN JavaScript/TypeScript. MDN is not enough. Correction done.